test/byContract/IexecPoco/04_finalize.test.ts

```diff
@@ -356,6 +356,103 @@ describe('IexecPoco2#finalize', async () => {
         expect(task.resultsCallback).to.equal('0x'); // deal without callback
     });
 
+    it('Should finalize task when callback address is EOA', async () => {
+        const callbackEOAAddress = ethers.Wallet.createRandom().address;
+        const orders = buildOrders({
+            assets: ordersAssets,
+            requester: requester.address,
+            prices: ordersPrices,
+            callback: callbackEOAAddress,
+        });
+
+        const { dealId, taskId, taskIndex } = await iexecWrapper.signAndMatchOrders(
+            ...orders.toArray(),
+        );
+        await iexecPoco.initialize(dealId, taskIndex).then((tx) => tx.wait());
+        const workerTaskStake = await iexecPoco
+            .viewDeal(dealId)
+            .then((deal) => deal.workerStake.toNumber());
+        const { resultHash, resultSeal } = buildResultHashAndResultSeal(
+            taskId,
+            callbackResultDigest,
+            worker1,
+        );
+        const schedulerSignature = await buildAndSignContributionAuthorizationMessage(
+            worker1.address,
+            taskId,
+            emptyEnclaveAddress,
+            scheduler,
+        );
+        await iexecWrapper.depositInIexecAccount(worker1, workerTaskStake);
+        await iexecPoco
+            .connect(worker1)
+            .contribute(
+                taskId,
+                resultHash,
+                resultSeal,
+                emptyEnclaveAddress,
+                emptyEnclaveSignature,
+                schedulerSignature,
+            )
+            .then((tx) => tx.wait());
+        await iexecPoco
+            .connect(worker1)
+            .reveal(taskId, callbackResultDigest)
+            .then((tx) => tx.wait());
+        await expect(iexecPocoAsScheduler.finalize(taskId, results, resultsCallback)).to.emit(
+            iexecPoco,
+            'TaskFinalize',
+        );
+    });
+
+    it('Should finalize task when callback address is non-EIP1154 contract', async () => {
+        const orders = buildOrders({
+            assets: ordersAssets,
+            requester: requester.address,
+            prices: ordersPrices,
+            callback: appAddress, // Non-EIP1154 contract
```

Comment:
Maybe use another random address to avoid confusion.

Reply:
here the appAddress was to meant to have a smart contract address and not a EOA without code in it

Reply:
How about that
https://github.com/iExecBlockchainComputing/PoCo/blob/482137414f3bcc37d64e6d6362594edc042cdf39/test/byContract/IexecPoco/IexecPoco1.test.ts#L121-L124

Or at least add a comment to explain than it's a random contract.